framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/scenario/ClusteredWorkbenchKieServerDatabasePersistentScenarioImpl.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.cloud.openshift.scenario;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.kie.cloud.api.deployment.DatabaseDeployment;
+import org.kie.cloud.api.deployment.Deployment;
+import org.kie.cloud.api.deployment.KieServerDeployment;
+import org.kie.cloud.api.deployment.SmartRouterDeployment;
+import org.kie.cloud.api.deployment.WorkbenchDeployment;
+import org.kie.cloud.api.deployment.constants.DeploymentConstants;
+import org.kie.cloud.api.scenario.ClusteredWorkbenchKieServerDatabasePersistentScenario;
+import org.kie.cloud.api.scenario.KieServerWithExternalDatabaseScenario;
+import org.kie.cloud.openshift.constants.OpenShiftTemplateConstants;
+import org.kie.cloud.openshift.deployment.DatabaseDeploymentImpl;
+import org.kie.cloud.openshift.deployment.KieServerDeploymentImpl;
+import org.kie.cloud.openshift.deployment.WorkbenchDeploymentImpl;
+import org.kie.cloud.openshift.template.OpenShiftTemplate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClusteredWorkbenchKieServerDatabasePersistentScenarioImpl extends OpenShiftScenario implements ClusteredWorkbenchKieServerDatabasePersistentScenario {
+
+    private WorkbenchDeploymentImpl workbenchDeployment;
+    private KieServerDeploymentImpl kieServerDeployment;
+    private DatabaseDeploymentImpl databaseDeployment;
+
+    private Map<String, String> envVariables;
+
+    private static final Logger logger = LoggerFactory.getLogger(KieServerWithExternalDatabaseScenario.class);
+
+    public ClusteredWorkbenchKieServerDatabasePersistentScenarioImpl(Map<String, String> envVariables) {
+        this.envVariables = envVariables;
+    }
+
+    @Override
+    public WorkbenchDeployment getWorkbenchDeployment() {
+        return workbenchDeployment;
+    }
+
+    @Override
+    public KieServerDeployment getKieServerDeployment() {
+        return kieServerDeployment;
+    }
+
+    @Override
+    public DatabaseDeployment getDatabaseDeployment() {
+        return databaseDeployment;
+    }
+
+    @Override public void deploy() {
+        super.deploy();
+
+        logger.info("Processing template and creating resources from " + OpenShiftTemplate.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getTemplateUrl().toString());
+        envVariables.put(OpenShiftTemplateConstants.IMAGE_STREAM_NAMESPACE, project.getName());
+        project.processTemplateAndCreateResources(OpenShiftTemplate.CLUSTERED_WORKBENCH_KIE_SERVER_DATABASE_PERSISTENT.getTemplateUrl(), envVariables);
+
+        workbenchDeployment = new WorkbenchDeploymentImpl(project);
+        workbenchDeployment.setUsername(DeploymentConstants.getWorkbenchUser());
+        workbenchDeployment.setPassword(DeploymentConstants.getWorkbenchPassword());
+        workbenchDeployment.scale(1);
```

Comment:
If workbench is clustered, shouldn't we want default count of replicas > 1 ?

Reply:
I set it to one to keep the same approach in all scenarios. That way all the tests can expect the same configuration available.

Reply:
Ok, make sense